File: apps/api/schemas/job.py
"""Job schemas for background task queue."""

from datetime import datetime
from uuid import UUID

from apps.api.schemas.base import BaseSchema, PaginatedResponse


class JobResponse(BaseSchema):
    """Single job status response."""

    id: UUID
    job_type: str
    status: str
    progress: int
    progress_message: str | None = None
    product_id: UUID | None = None
    user_id: str | None = None
    input_data: dict | None = None
    result_data: dict | None = None
    error_message: str | None = None
    started_at: datetime | None = None
    completed_at: datetime | None = None
    retry_count: int = 0
    arq_job_id: str | None = None
    created_at: datetime
    updated_at: datetime


class JobCreateResponse(BaseSchema):
    """Response returned when a job is enqueued."""

    job_id: UUID
    status: str


class JobListResponse(PaginatedResponse):
    """Paginated job list."""

    data: list[JobResponse]

File: apps/web/src/lib/types/job.ts
export type JobStatus = "pending" | "running" | "completed" | "failed";

export interface Job {
  id: string;
  job_type: string;
  status: JobStatus;
  progress: number;
  progress_message: string | null;
  product_id: string | null;
  user_id: string | null;
  input_data: Record<string, unknown> | null;
  result_data: Record<string, unknown> | null;
  error_message: string | null;
  started_at: string | null;
  completed_at: string | null;
  retry_count: number;
  arq_job_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface JobCreateResponse {
  job_id: string;
  status: string;
}
